lib/SurveyGizmo.js

```diff
@@ -59,29 +62,28 @@ class SurveyGizmo {
   }
 
   /**
-   * Renders data as an object.
-   * @returns {object} containing valuable response attributes.
+   * Tangles question data with response data to provide a useful data format.
+   * @param {array} responses
+   * @returns {array} of objects that have id, submittedAt and questions fields.
+   *                  Questions field contains key: value pairs where key is
+   *                  SurveyGizmo question title and value is person's response value.
    */
-  getData() {
-    return this.getQuestions()
-      .then(() => this.getResponses())
-      .then(data =>
-        data.map((item) => {
-          let questions = {};
-          Object.keys(item).map(key => {
-            let question = this.getQuestion(key);
-            if(question && question.title) {
-              questions[question.title.English] = item[key];
-            }
-          });
+  tangleQuestionsAndResponses(responses) {
```

Comment:
Much better way would be if this function would take reponses and questions as it's arguments, then zip them with zipWith (lodash/underscore/...).